test/unit/api/define.js

```diff
@@ -3,18 +3,38 @@ import { Component, define, symbols } from '../../../src/index';
 const { name: $name } = symbols;
 
 describe('api/define', () => {
+  it('should not register without any properties', () => {
+    expect(() => {
+      define('x-test');
+    }).to.throw(Error);
+  });
+
+  it('should not register a component without a dash in the name', () => {
```

Comment:
Maybe you're right, it just looks nicer for me to see all the limitations in the one place. I'll remove it if you sure this is not worth is.